### PR TITLE
#6: add multi-select feature to fuzzy-search

### PR DESCRIPTION
**Pull request by lavgl**
What's rendering for item that both current and marked is a subject for discussion :)

Also, you can see several pure functions in actions/fuzzy-search.js file, which can be easy unit-tested. Should i do that?

### CODE REVIEW

**Comment by ranyitz** [comment]
Wow, that's really nice! 🎉 

Regarding the `current&marked` state.

screen shot 2018-03-26 at 21 48 20

I think we should preserve the position of the string so it won't jump from left to right, also maybe let's do the marked in a different color, for example `magenta` (instead of taking it one step backward).

What do you think?

**Comment by lavgl** [comment]
Thanks :)

Agree, moving parts of UI backward/forward could be annoying for users :)

Do you mean something like this?

screen shot 2018-03-27 at 12 01 26 am

**Comment by ranyitz** [comment]
What do you say about something more similar to this?

screen shot 2018-03-27 at 21 14 22

When the `marked` cursor is indented differently than the `current` one?
The text should stay the same, only the cursor should be in a different color.

File: src/actions/fuzzy-search.js [hunk 1]
```diff
@@ -81,15 +128,45 @@ module.exports = workspace => {
     renderUi();
   });
 
+  input.on('tab', () => {
+    const { value } = results[currentResult];
```
Comment (ranyitz) [inline src/actions/fuzzy-search.js]: What about a case when `results.length === 0` ?

You'll try to destructure value from an `undefined` which will raise an error.

File: src/actions/fuzzy-search.js [hunk 2]
```diff
@@ -81,15 +128,45 @@ module.exports = workspace => {
...
+
+    if (currentResult > 0) {
+      currentResult--;
+    }
+
+    renderUi();
+  });
+
   input.on('choose', () => {
     if (results.length === 0) {
```
Comment (ranyitz) [inline src/actions/fuzzy-search.js]: If we return on `results.length === 0` it means that even if we chose some modules we can't press enter :)

Maybe we should also check if there are no chosen modules.

`results.length === 0 && chosen.length === 0`

**Comment by lavgl** [comment]
thanks for comments, i will reiterate on it. 

also, i think, it make sense to always display marked items.

consider the situation:
1. user enters fuzzy-search mode
2. user types, for example, `lodash` and see the list [`lodash`, `lodash.uniq`, `lodash.get`]
3. user marks `lodash` and appends `.g`, so input string becomes `lodash.g`. `lodash` will be filtered out.

but as for me, it makes more sense to show it

what do you think?

**Comment by ranyitz** [comment]
@lavgl would you be interested in adding some explanations about how to work with the watch mode in the README?

Something like [this](https://github.com/junegunn/fzf#using-the-finder) maybe?

**Comment by lavgl** [comment]
@ranyitz thanks for review! sure, will add some docs regarding watch mode

**Pull request merged**